### ISSUE TITLE
Rel styling $lineStyle does not work for sequence diagrams

### ISSUE THREAD

**Issue by papa-vova** [plantuml-stdlib/C4-PlantUML#369]
Hi! Thank you for a super awesome tool.

I have a line styling issue with sequence diagram.

The code:
```plantuml
@startuml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Sequence.puml
AddRelTag("response", $lineStyle=DashedLine())

Container(a1, "a1")
Container(a2, "a2")
Rel(a1, a2, "request")
Rel(a2, a1, "response", $tags="response")

@enduml
```

produces the following output:
![rel styling bug](https://github.com/user-attachments/assets/1ede53e7-0211-41d2-9e47-8bbea1ad10d8)

I would expect the response line to be dashed. Am I doing something wrong? can't google up anything helpful.

I am not sure if the line «Only following (extended) relationship specific macros is supported:» from the [README](https://github.com/plantuml-stdlib/C4-PlantUML/blob/master/README.md#c4-styled-sequence-diagram) forbids styling like this.

For container diagrams the same styling works perfectly well.

Thanks!

p.s.
![rel styling bug](https://github.com/user-attachments/assets/ebba5570-d460-4c8b-b85d-13b4e2f37ad2)

**Comment by papa-vova** [comment]
also, when i change the include to the following:
```plantuml
!include <C4/C4_Sequence>
```

I get an error:
![rel styling bug](https://github.com/user-attachments/assets/26c9da6d-42f0-4d59-ad95-78007e7695d3)

but i guess it's another issue...

**Comment by kirchsth** [comment]
Hi @papa-vova,

A) related to `!include <C4/C4_Sequence>`:
     I cannot reproduce the issue on the ex. online server.
     
[![https://www.plantuml.com/plantuml/uml/LS-noi8m50NWFK-HF_n30WLfoMYb4aSdAwuIcqiDfAdctWvzUzE2g6E-So3dDaWsqJG4zkTZ5oO7V6lqnkXh2yy9OWSxnlREdI2SRIz50doC4K4K_3_u22tD0Ug3nJkuOtvBfHXZxOm4W-mqaWl-Ld2ZBv3GZt7DpHZ9vdAIjYouiALGtrOjLcNBdwwDLTAo1-dDcPRePn3P7kkFADP0TFcs5m00](https://www.plantuml.com/plantuml/png/LS-noi8m50NWFK-HF_n30WLfoMYb4aSdAwuIcqiDfAdctWvzUzE2g6E-So3dDaWsqJG4zkTZ5oO7V6lqnkXh2yy9OWSxnlREdI2SRIz50doC4K4K_3_u22tD0Ug3nJkuOtvBfHXZxOm4W-mqaWl-Ld2ZBv3GZt7DpHZ9vdAIjYouiALGtrOjLcNBdwwDLTAo1-dDcPRePn3P7kkFADP0TFcs5m00)](https://www.plantuml.com/plantuml/uml/LS-noi8m50NWFK-HF_n30WLfoMYb4aSdAwuIcqiDfAdctWvzUzE2g6E-So3dDaWsqJG4zkTZ5oO7V6lqnkXh2yy9OWSxnlREdI2SRIz50doC4K4K_3_u22tD0Ug3nJkuOtvBfHXZxOm4W-mqaWl-Ld2ZBv3GZt7DpHZ9vdAIjYouiALGtrOjLcNBdwwDLTAo1-dDcPRePn3P7kkFADP0TFcs5m00)
     
    A1) Can you reproduce it online and add the link
    A2) If only offline, can you past the whole sample (maybe there is a typo,...?)
    
B) related to  `$lineStyle=DashedLine()`:  I have to check , maybe it was never working  
   
BR Helmut

**Comment by papa-vova** [comment]
A) hmmm online [everything works](https://www.plantuml.com/plantuml/uml/LSx12u8m5CVn-_uA6nqKX50yPXXsxAJTu-5-cB1cxJqF_VTj4bJ7tsU3xsjOoCjojsepkS4k1ic-hNPjTUtnNE067DJHc0xsGcEgFVWnEuREawsT77fvMTGduXlCEUmqovHgPoSKXa-fo1DDXSv-hOnM1kjWrn-h-9XZ-N2a-FGN51gv_aeeDN0ctFu6)

here's the code:
```plantuml
@startuml
!include <C4/C4_Sequence>
AddRelTag("response", $lineStyle=DashedLine())

Container(a1, "a1")
Container(a2, "a2")
Rel(a1, a2, "request")
Rel(a2, a1, "response", $tags="response")

@enduml
```

**Comment by kirchsth** [comment]
Can you create attach your *.puml file and create a *.preproc file and attach it too (e.g. if your file is "sequence.puml" then you can generate it with following line)

```bash
java -jar plantuml-mit-1.2024.7.jar ./sequence.puml

java -jar plantuml-mit-1.2024.7.jar -preproc ./sequence.puml
```

**Comment by papa-vova** [comment]
[rel styling bug.zip](https://github.com/user-attachments/files/17422704/rel.styling.bug.zip)

**Comment by kirchsth** [comment]
Your attached *.preproc file is too small with 1.6kB (many lines are missing, but I don't know why); my local has ~39kB.
Maybe PlantUML has problems with your used codepage or an invisible char (but I didn't found one).
(the *.preproc file itself is a valid *.puml file, only all preprocessor commands are resolved)

Can you compile my file?

[workingOnMyMachine.zip](https://github.com/user-attachments/files/17423118/workingOnMyMachine.zip)
 
If my file is working on your machine too, I would retype your file manually and check if the new *.puml is working.

PS.: I attached my *.preproc output from your file too that you can see the expected result

**Comment by kirchsth** [comment]
For the dotted line problem I have a workaround (but maybe I find a final solution over the weekend)

```plantuml
@startuml
!include <C4/C4_Sequence>

 ' defines a workaround arrow, with the a logical matching name "$response" that it can be used by the $rel argument
!$response = "-->"
....
Rel(a2, a1, "response", $rel = $response)
````

full sample
 
```plantuml
@startuml
!include <C4/C4_Sequence>

' defines a woraround arrow, with the a logical matching name "$response" that it can be used by the $rel argument
!$response = "-->"

AddRelTag("response", $textColor=$ARROW_COLOR, $lineColor=#red, $lineStyle=DashedLine())


System(c4stdlib, C4Version())

Container(a1, "a1")
Container(a2, "a2")
Rel(a1, a2, "request")

' legend not working legend
Rel(a2, a1, "response", $rel = $response)

Rel(a1, a2, "request2", $rel = $response)


' for a working legend you have to define the $tags too
Rel(a2, a1, "response2", $tags="response", $rel = $response)

SHOW_LEGEND()
@enduml
```
 
[![](https://www.plantuml.com/plantuml/png/ZP1HIyCm58NVxw_8ui0MDcIZZtPiP4CVXeLMz774vj85qqIJM-R-lMatHM6WZ_dEEJcSkmmaFNMjWP6sjUaKibkHtOXiL-5xXxR61S1ex36yEHkGvOnFfmiEi5AgHFCecuH_YNp2neGV99nnFX-loh9utebYMvHHCDhYIRZogCwWegF1V2t37jKslfCq1O3g60ZRfCu2AQDV9anaJ-Y3TdRGXRCaezadSZPXNCvu-fFDUpQFh4GpE0RY-pM18WQuPWORj8fPH-pW_AkspHaDeJxG1tyDysZY-Ez3f73n__a5wwdoe6d_hxBX2v9DoF_eX-e-tdUxkTiyh9CKbcXLrvfF)](https://www.plantuml.com/plantuml/uml/ZP1HIyCm58NVxw_8ui0MDcIZZtPiP4CVXeLMz774vj85qqIJM-R-lMatHM6WZ_dEEJcSkmmaFNMjWP6sjUaKibkHtOXiL-5xXxR61S1ex36yEHkGvOnFfmiEi5AgHFCecuH_YNp2neGV99nnFX-loh9utebYMvHHCDhYIRZogCwWegF1V2t37jKslfCq1O3g60ZRfCu2AQDV9anaJ-Y3TdRGXRCaezadSZPXNCvu-fFDUpQFh4GpE0RY-pM18WQuPWORj8fPH-pW_AkspHaDeJxG1tyDysZY-Ez3f73n__a5wwdoe6d_hxBX2v9DoF_eX-e-tdUxkTiyh9CKbcXLrvfF)

**Comment by kirchsth** [comment]
Based on following check, only the `$lineStyle` is not working. All other works

```
@startuml
!include <C4/C4_Sequence>

AddRelTag("response", $textColor="#green", $lineColor="red", $sprite="&envelope-closed", $techn="HTTPS", $legendSprite="&envelope-open", $legendText="response\nchanged", $lineStyle=DottedLine())

System(c4stdlib, C4Version())

Container(a1, "a1")

Rel(a1, a2, "request")
Rel(a2, a1, "response", $tags="response")

SHOW_LEGEND()
@enduml
```

[![](https://www.plantuml.com/plantuml/svg/PP1DImCn68Jl-HLh5Db2HIfxD5BP5dieAjr5BqA9oR1TID_Kv5sn_zwaVXJnaCCyJyA4cKRMWOUj4oSz6JTO5DTrTLbNwmPl0yZWHeXRQrTmhUvA6H1tdYBak3XZV73jdGzAddO1e0nTJ_Y60JQJk0izGybpq3kSt-72E1-_7CDiICb5spusEJjq8DlyUv4E7NsRYjNlLrx8R3Hri3_r3UyTrCmpmovJBaSZ8PfzP6nBKqMshdyT5tNrX11xJmTTUsATxePIJyQ5r1EPu0hk4FLLGY7l4JdnZ3FAwkyWkelgI7Bhuk5vlPpVpUzdvKXCGJPD_Ga0)](https://www.plantuml.com/plantuml/uml/PP1DImCn68Jl-HLh5Db2HIfxD5BP5dieAjr5BqA9oR1TID_Kv5sn_zwaVXJnaCCyJyA4cKRMWOUj4oSz6JTO5DTrTLbNwmPl0yZWHeXRQrTmhUvA6H1tdYBak3XZV73jdGzAddO1e0nTJ_Y60JQJk0izGybpq3kSt-72E1-_7CDiICb5spusEJjq8DlyUv4E7NsRYjNlLrx8R3Hri3_r3UyTrCmpmovJBaSZ8PfzP6nBKqMshdyT5tNrX11xJmTTUsATxePIJyQ5r1EPu0hk4FLLGY7l4JdnZ3FAwkyWkelgI7Bhuk5vlPpVpUzdvKXCGJPD_Ga0)

**Comment by papa-vova** [comment]
> If my file is working on your machine too, I would retype your file manually and check if the new *.puml is working.

hmmm
sequence.puml does `not` work for me (

strange. i have notepad++ as an editor, and the box in win10. the encoding is utf-8
![image](https://github.com/user-attachments/assets/c4ee2efb-082a-4c3f-81ea-fc9f39519eff)

> Based on following check, only the $lineStyle is not working. All other works

yes i also checked colors & stuff before posting, only the line style seems to be off

**Comment by papa-vova** [comment]
> I have a workaround

yes it works for me (quoting the original file where i've got this problem)

![image](https://github.com/user-attachments/assets/19d92e0b-74e8-4b36-a73a-085b7926d5ae)

thanks! owe you one )

**Comment by papa-vova** [comment]
btw 

> [full sample](https://github.com/plantuml-stdlib/C4-PlantUML/issues/369#issuecomment-2420657832)

i realized that stuff like `$textColor=$ARROW_COLOR` would not work as well, but `$textColor=#red` works

**Comment by kirchsth** [comment]
In the meantime I think you have a problem in your (downloaded) *.jar file or a problem with the encoding of your *.puml.
a) can you try to store your *.puml as ANSI 
b) can you try it with an other/older, ex. plantuml-mit-1.2024.**6**.jar

PS.: `$textColor=$ARROW_COLOR` does no really change the color - because it is the same, it avoids only the "last text color" in the default text of the legend ![image](https://github.com/user-attachments/assets/a46489e6-3cf9-4960-a4fd-2e2c5fbfa22d)
Ex. with `$textColor=$LEGEND_DARK_COLOR` or `$textColor=$SKETCH_WARNING_COLOR`  you should see the difference

**Comment by papa-vova** [comment]
> can you try to store your *.puml as ANSI

```shell
papavova@Thinker:/mnt/c/Users/papa.vova/Downloads$ file rel\ styling\ bug.puml
rel styling bug.puml: ASCII text, with CRLF line terminators
```

does not seem to help (

> can you try it with an other/older, ex. plantuml-mit-1.2024.6.jar

does not seem to help either (( error's exactly the same as far as i can tell:

![image](https://github.com/user-attachments/assets/d16f4b72-0561-4f00-ba05-dfab82fcede9)
```plantuml
@startuml
!include <C4/C4_Sequence>

Container(a1, "a1")
Container(a2, "a2")
Rel(a1, a2, "request")
Rel(a2, a1, "response")

@enduml
```

HOWEVER! i *did* have a jar that was 11Mb instead of 21+! that's what win explorer showed to me anyway. mystery how it even worked. maybe i should fully move to wsl or linux (

thanks anyway
i think it's a problem on my side, will try to figure it out
please do not bother any more. sorry (

**Comment by The-Lum** [comment]
Hi @papa-vova, @kirchsth 

Here is a clue, from your screenshot:

>![image](https://github.com/user-attachments/assets/415b05a8-c1a6-45a9-b713-b11c2312ffed)

- What are the characters before `!include`?

_If that can help,_
Regards,
Th.

**Comment by papa-vova** [comment]
Hi @The-Lum ! Yes I've noticed that, but I can't spot those characters anywhere.

Well anyway, here's what i did now:
1) fully retyped the original file (with Sequence)
2) got the same error
3) replaced Sequence with Container
4) got a proper diagram

(PlantUML runs on Win, the shell is on WSL2 which does not have Java, sorry about the mess)

![image](https://github.com/user-attachments/assets/929fb9d3-6128-4dd6-b293-3fd29e63e024)

**Comment by kirchsth** [comment]
I still think it is the *.jar file. Could it be that your downloaded *.zip or the unpacked *.puml is blocked?''
https://superuser.com/questions/590787/what-does-it-mean-when-a-file-is-blocked-in-windows 

PS.: And yes, If the size of your *.jar file is wrong then the content is wrong and then also the internal stored <C4/C4:....*> could have an issue.

**Comment by papa-vova** [comment]
Oh man, I thought they would not let me move blocked files around, but apparently they do.
The file _was_ blocked, but unblocking did not change the outcome (

The size seems to be correct.

I am certain the issue is with the jar, i just can't spot it.

Could anybody point me to the manual as to how to check the `.asc` signature?

**Comment by kirchsth** [comment]
you could use certutil, it support multiple hash codes 
https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/certutil
(supported hash algorithms: MD2 MD4 MD5 SHA1 SHA256 SHA384 SHA512 )

e.g. `certutil -hashfile yourFile.jar SHA256`

**Comment by papa-vova** [comment]
They [don't seem to have](https://github.com/plantuml/plantuml/releases/tag/v1.2024.7) anything except asc pgp stuff. All the googling [points to gpg](https://serverfault.com/questions/896228/how-to-verify-a-file-using-an-asc-signature-file), so maybe i should bite the bullet (and find out where's the public key etc)

I am not sure how to use `certutil`, maybe this is something really straightforward but i'm a bit ~old~ new to stis stuff, here's what i get:

![image](https://github.com/user-attachments/assets/a065bbf1-b7c5-4386-b0e1-cd72505a17cc)

**Comment by kirchsth** [comment]
I think "certutil" is not the correct tool, therefore I deleted my comment.
But if you have a wrong file size then the file is wrong independent of the signatur.

Did you try to download the *.zip again, **unblock the *.zip**  and then try to extract the file maybe then you have a working version

**Comment by papa-vova** [comment]
downloaded the jar [from github](https://github.com/plantuml/plantuml/releases/tag/v1.2024.7), unblocked, moved around, tried to rebuild — still no luck

i still think that win is messing with the file somehow

either that, or the include file for the sequence has some obscure bug which shows only on windows (?)

for the size, they have literally 21.1Mb which this time looks not incorrect )

**Comment by kirchsth** [comment]
I checked it with win10 too->no problems; Maybe you have to unblock the *.jar file too

**Comment by papa-vova** [comment]
that's the only thing i have: the jar file
no zip archives

**Comment by kirchsth** [comment]
Sorry you are right, the *.zip was the file which you gave me.
If you still get the same *preproc file then (I think) still the old *.jar is used, otherwise I would expect a difference. 

But back to the original problem: I found already an [issue in the forum that sequence diagrams do not support line styles](https://forum.plantuml.net/15612/style-sequence-allow-linestyle-for-arrow-sequence-diagram?show=15612#q15612), maybe I get a fast bugfix and then I will update my implementation too.

**Comment by arnaudroques** [comment]
> But back to the original problem: I found already an [issue in the forum that sequence diagrams do not support line styles](https://forum.plantuml.net/15612/style-sequence-allow-linestyle-for-arrow-sequence-diagram?show=15612#q15612), maybe I get a fast bugfix and then I will update my implementation too.

The https://github.com/plantuml/plantuml/commit/64ea23246d59a76e6bf7a1ed01e57764b090d0e4  commit may fix it...
Please test it and tell us :-)

**Comment by kirchsth** [comment]
Hi @arnaudroques,
thank you, yes it works perfect; Even backward compatible with (old) skinparam I don't need to change anything in my code.
BR Helmut
PS.: I will update the forum entry too

**Comment by kirchsth** [comment]
Hi @papa-vova,

$lineStyle works with the last beta.

If you download the latest PlantUML beta v1.2024.8beta5 [plantuml-mit-SNAPSHOT.jar](https://github.com/plantuml/plantuml/releases/download/snapshot/plantuml-mit-SNAPSHOT.jar)  **(and unblock it)** then you can test it locally on your machine. (I hope you have no old *.jar file with the same name on your local system , if yes I would try a different license that you have a file with a complete new name).

I hope it works for you too. (otherwise you have to wait until the online server is updated; you can check it with following code - the brackets creates a new component and its label is the version number) 
  
```plantuml
@startuml
[%version()]
@enduml
```

BR Helmut

**Comment by papa-vova** [comment]
Whoa! thanks )))

![rel styling bug](https://github.com/user-attachments/assets/ac75afe8-c69c-4d4d-b890-6777d165fdb5)

the code (the variant with `!include <C4/C4_Sequence>` still fails):

```plantuml
@startuml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Sequence.puml
AddRelTag("response", $lineStyle=DashedLine())

Container(a1, "a1")
Container(a2, "a2")
Rel(a1, a2, "request")
Rel(a2, a1, "response", $tags="response")

@enduml
```

**Comment by kirchsth** [comment]
It is fixed with PlantUML v1.2024.8 (beta5)
